core/array.rbs

```diff
@@ -1736,7 +1736,7 @@ class Array[unchecked out Elem] < Object
   # See also Enumerable#sort_by.
   #
   def sort: () -> ::Array[Elem]
-          | () { (Elem a, Elem b) -> ::Integer? } -> ::Array[Elem]
+          | () { (Elem a, Elem b) -> ::Integer } -> ::Array[Elem]
```

Comment:
`Array#sort`'s block doesn't accept nil as the returned value actually, so I removed nil.


It is a more strict type, but it can be problematic with `<=>` method.
For example, `Integer#<=>` may return a `nil` when the values are incomparable. So the returned type is `Integer?`.
https://github.com/ruby/rbs/blob/26baa08e184d1d27c18a34e6683493db42f5fe77/core/integer.rbs#L108-L115

So `Array#sort` will not become to accept the following code by this change.

```ruby
[1,2,3].sort { |a, b| a <=> b }
```


But personally I think `Integer` is better than `Integer?` as the returned value of `sort`'s block. `Array#sort` always raises when the block returns nil.
And we can improves `Integer#<=>` type, like `(Integer) => Integer | (Numeric) -> Integer?`. The improvement will resolve the `Array#sort`'s problem.

Reply:
This makes sense. 👍 

(I made it `Integer?` because of the return type of `<=>`.)